src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -239,10 +242,10 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
             TaskUtilities.AssertIsOnBackgroundThread();
 
             _callback = callback;
-            _startupInfo = startupInfo;
+            _startupInfo = startupInfo ?? new RHostStartupInfo();
             RHost host;
             try {
-                var connectionInfo = new BrokerConnectionInfo(startupInfo.Name, this, startupInfo.RHostCommandLineArguments, timeout);
+                var connectionInfo = new BrokerConnectionInfo(Name, this, startupInfo.UseRHostCommandLineArguments, timeout);
```

Comment:
command line args is a connection level setting. Individual sessions may or may not want to use arguments. REPL session uses args, Package Manager session doesn't need them. I'm not sure about intellisense session: command line args may contain a script that loads a package. Does it matter?